src/networking/nc/mod.rs

```diff
@@ -0,0 +1,958 @@
+extern crate mio;
+extern crate clap;
+extern crate libc;
+extern crate socket2;
+
+use std;
+use failure;
+use tempfile::NamedTempFile;
+use clap::{Arg, ArgMatches};
+use mio::{Events, Event, Poll, Ready, PollOpt, Token};
+use libc::{AF_UNSPEC, AF_INET, AF_INET6, AF_UNIX};
+use std::io;
+use std::net::{SocketAddr};
+use mio::unix::EventedFd;
+use std::io::{Read,Write, ErrorKind};
+use mio::unix::UnixReady;
+use std::thread::sleep;
+use std::fmt::Debug;
+use std::os::unix::io::AsRawFd;
+use std::os::unix::io::RawFd;
+use std::time::Duration;
+use std::net::{ToSocketAddrs};
+use socket2::{Socket, Domain};
+use super::{UtilSetup, ArgsIter};
+use super::{MesaError};
+
+
+pub(crate) const NAME: &str = "nc";
+pub(crate) const DESCRIPTION: &str = "arbitrary TCP and UDP connections and listens";
+
+const BUFSIZE: usize = 16384;
+const PRINT_DEBUG_INFO: bool = false;
+
+#[derive(Debug)]
+struct NcOptions {
+    dflag: bool,
+    iflag: bool,
+    interval: Option<Duration>,
+    lflag: bool,
+    host: String,
+    portlist: Vec<u16>,
+    unixflag: bool,
+    uflag: bool,
+    family: i32,
+    kflag: bool,
+    s_addr: Option<String>,
+    pflag: bool,
+    vflag: bool,
+    zflag: bool,
+    timeout: Option<Duration>,
+    unix_dg_tmp_socket: String,
+}
+
+fn mesaerr_result<T>(err_msg: &str) -> Result<T, MesaError> {
+    Err(MesaError::from(
+        failure::err_msg(format!("{}", err_msg)).compat()
+        ))
+}
+
+fn build_ports(ports: &str) -> Result<Vec<u16>, MesaError>{
+    // TODO: suport XX-XX
+    let port_list = match ports.parse::<u16>() {
+        Ok(port) => port,
+        Err(_) => {
+            return mesaerr_result(&format!("invalid port[s] {}", ports));
+        }
+    };
+
+    Ok(vec!(port_list))
+}
+
+fn warn(msg: &str) {
+    eprint!("{}", msg);
+}
+
+fn debug_info(msg: &str) {
+    if PRINT_DEBUG_INFO {
+        eprint!("{}", msg);
+    }
+}
+
+impl NcOptions {
+    pub fn parse(matches: ArgMatches, msg: &str) -> Result<NcOptions, MesaError> {
+        let mut portlist = vec!();
+        let lflag = matches.is_present("l");
+        let mut host = String::from("127.0.0.1");
+        let uport:String;
```

Comment:
There should be a space after `:`.
 - `let uport: String;`

Reply:
done